user-service/src/main/java/com/ecommercepractice/userservice/dao/FakeUserDataAccessService.java

```diff
@@ -0,0 +1,58 @@
+package com.ecommercepractice.userservice.dao;
+
+import com.ecommercepractice.userservice.models.User;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository("fakeDao")
+public class FakeUserDataAccessService implements IUserDao {
+    private static List<User> DB = new ArrayList<User>();
+
+    @Override
+    public int insertUser(User user) {
+        DB.add(user);
+        return 1;
+    }
+
+    @Override
+    public List<User> selectAllUsers() {
+        return DB;
+    }
+
+    @Override
+    public Optional<User> selectUserByUserName(String userId) {
```

Comment:
Should not return an Optional, should Return a User o throws an Exception indicating that the information was not found

Reply:
I propose DAO returns Optional and Service handle the exception